### PR TITLE
Features/coin control stage3

### PR DESCRIPTION
**Pull request by SuperJMN**

### CODE REVIEW

File: WalletWasabi.Fluent/Converters/IndentConverter.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+using Avalonia;
+using Avalonia.Data.Converters;
+
+namespace WalletWasabi.Fluent.Converters;
+
+public class IndentConverter : AvaloniaObject, IValueConverter
+{
+	public static readonly StyledProperty<double> MultiplierProperty = AvaloniaProperty.Register<IndentConverter, double>(nameof(Multiplier));
+
+	public static Avalonia.Controls.Converters.IndentConverter Instance { get; } = new();
```
Comment (RolandUI) [inline WalletWasabi.Fluent/Converters/IndentConverter.cs]: I don't get this. Due to `Avalonia.Controls.Converters` this doesn't create an instance from your class. Is this on purpose?

File: WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	private static IColumn<CoinControlItemViewModelBase> AmountColumn()
+	{
+		return new PlainTextColumn<CoinControlItemViewModelBase>(
+			"Amount",
+			node => node.Amount.ToFormattedString(),
+			GridLength.Auto,
+			new ColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, Money>(x => x.Amount),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, Money>(x => x.Amount)
```
Comment (RolandUI) [inline WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs:127]: null warnings

File: WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	private static IColumn<CoinControlItemViewModelBase> IndicatorsColumn()
+	{
+		return new TemplateColumn<CoinControlItemViewModelBase>(
+			"",
+			new FuncDataTemplate<CoinControlItemViewModelBase>((_, _) => new IndicatorsCellView(), true),
+			GridLength.Auto,
+			new ColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, int>(GetIndicatorPriority),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, int>(GetIndicatorPriority)
```
Comment (RolandUI) [inline WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs:140]: null warnings

File: WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	private static IColumn<CoinControlItemViewModelBase> AnonymityScoreColumn()
+	{
+		return new PlainTextColumn<CoinControlItemViewModelBase>(
+			new AnonymityScoreHeaderView(),
+			node => node is CoinCoinControlItemViewModel coin ? coin.AnonymityScore.ToString() : "",
+			GridLength.Auto,
+			new TextColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, int?>(b => b.AnonymityScore),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, int?>(b => b.AnonymityScore)
```
Comment (RolandUI) [inline WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs:153]: null warnings

File: WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	private static IColumn<CoinControlItemViewModelBase> PocketColumn()
+	{
+		return new TemplateColumn<CoinControlItemViewModelBase>(
+			"Pocket",
+			new FuncDataTemplate<CoinControlItemViewModelBase>((_, _) => new LabelsCellView(), true),
+			GridLength.Star,
+			new ColumnOptions<CoinControlItemViewModelBase>
+			{
+				CompareAscending = SortAscending<CoinControlItemViewModelBase, int>(GetLabelPriority),
+				CompareDescending = SortDescending<CoinControlItemViewModelBase, int>(GetLabelPriority)
```
Comment (RolandUI) [inline WalletWasabi.Fluent/ViewModels/CoinControl/CoinSelectorViewModel.cs:166]: null warnings

File: WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs [hunk 6]
```diff
@@ -27,192 +21,41 @@ namespace WalletWasabi.Fluent.ViewModels.CoinControl;
...
-				pocketColumn
-			}
-		};
+		CoinSelector = new CoinSelectorViewModel(walletViewModel, selectedCoins);
 
-		Source.SortBy(pocketColumn, ListSortDirection.Descending);
-		Source.RowSelection!.SingleSelect = true;
+		var requiredAmount = CoinSelector.SelectedCoinsChanged.Select(GetRequiredAmount);
+		var selectedAmount = CoinSelector.SelectedCoinsChanged.Select(c => new Money(c.Sum(x => x.Amount)));
+		var remainingAmount = selectedAmount.CombineLatest(requiredAmount, (selected, remaining) => remaining - selected);
```
Comment (RolandUI) [inline WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs]:
```suggestion
		var remainingAmount = selectedAmount.CombineLatest(requiredAmount, (selected, required) => required - selected);
```

File: WalletWasabi.Fluent/Views/CoinControl/Core/Cells/SelectionCellView.axaml [hunk 7]
```diff
@@ -7,6 +7,5 @@
              x:Class="WalletWasabi.Fluent.Views.CoinControl.Core.Cells.SelectionCellView"
              x:DataType="core:CoinControlItemViewModelBase"
              x:CompileBindings="True">
-  <!-- TODO: Temporarily disabled until the final implementation -->
-  <CheckBox IsChecked="{Binding IsSelected}" IsVisible="{Binding CanBeSelected}" IsEnabled="False" />
+  <CheckBox IsChecked="{Binding IsSelected}" IsEnabled="{Binding CanBeSelected}" />
```
Comment (RolandUI) [inline WalletWasabi.Fluent/Views/CoinControl/Core/Cells/SelectionCellView.axaml]: There was not much difference between the disabled and enabled state of the checkbox, so we decided to go with IsVisible (as it is in the wallet coins view). Are we sure this won't annoy the users?

File: WalletWasabi.Fluent/Converters/IndentConverter.cs (hunk 1 above)
Comment (SuperJMN) [reply WalletWasabi.Fluent/Converters/IndentConverter.cs]: You mean the `Instance` property? I copied this converter from the TreeDataGrid project and forgot to delete it. It makes no sense here.

File: WalletWasabi.Fluent/Views/CoinControl/Core/Cells/SelectionCellView.axaml (hunk 7 above)
Comment (SuperJMN) [reply WalletWasabi.Fluent/Views/CoinControl/Core/Cells/SelectionCellView.axaml]: I thought it was a mistake. Sorry, didn't know we decided that.
I think it's a good decision. Let's go with that.

**Pull request closed without merging**